### ISSUE TITLE
Breaks xmodmap

### ISSUE THREAD

**Issue by dwrz** [oh-my-fish/theme-bobthefish#160]
I have custom keybindings defined with xmodmap (https://wiki.archlinux.org/index.php/Xmodmap).

They were working fine until I updated omf and fisherman. Then my keybindings stopped working.

Uninstalling bobthefish and re-sourcing my xmodmap config resulted in the bindings working again.

As soon as I install bobthefish, xmodmap stops working. 

Re-sourcing the config with bobthefish installed doesn't fix the keybindings.

I skimmed through the bobthefish files, but I'm lost as to what could be affecting xmodmap.

Are there any insights you could provide to help fix this issue?

**Comment by bobthecow** [comment]
Ooh, that's an interesting one.

If I understand it, the things you have in the mix are OMF, Fisherman and bobthefish?

bobthefish itself shouldn't be messing with keybindings at all, but it wouldn't surprise me if OMF or fisherman was, or if having both of them installed at the same time was part of the problem. You can test this by removing OMF and Fisherman, and installing the source files for bobthefish directly.

You should be able to add them to any folder in `$fish_function_path`… probably `$HOME/.config/fish/functions` is the easiest? Ignore the install hook, but just dump all the other `.fish` files from this repo directly in a folder in your function path and restart fish.

Let me know if that works?

**Comment by dwrz** [comment]
With neither the theme, omf, or fisher installed, adding the theme files to `$HOME/.config/fish/functions` results in xmodmap breaking. I am really lost as to why.

I also tried it with just fisher installed, though not with just omf installed. I'd be happy to try that if it would be of any use.

This is my config.fish:
```
set -g fish_greeting ''
set -g theme_color_scheme zenburn
set -g theme_powerline_fonts yes
set -g theme_nerd_fonts yes
set -g theme_display_vi no
set -g theme_display_date yes
set -g theme_date_format "+%H%M%S"
set -g theme_title_display_process yes
set -g theme_display_user yes
```

Thank you for the help and the quick response, too!

**Comment by bobthecow** [comment]
hmm. that is strange. let's try including just one file at a time? you'll need __bobthefish_colors and __bobthefish_glyphs for fish_prompt, but otherwise the rest of the files should be independent.

**Comment by dwrz** [comment]
fish_prompt.fish seems to be the culprit. As soon as I add it, the shell changes visually (it's a lot prettier), but xmodmap breaks. : \

**Comment by bobthecow** [comment]
Unfortunately that’s not super helpful, since that’s the bulk of the code in this theme :P

You up for some surgery? Open that file and comment out every line in the guts of the main function (named fish_prompt). Restart fish… you should have no visible prompt, but maybe xmodmap?

If that works, uncomment one line at a time to see which section is breaking it?

**Comment by dwrz** [comment]
Haha, thank you. It's __bobthefish_prompt_rubies. With that line uncommented, everything seems to work.

**Comment by bobthecow** [comment]
Great! Thanks for helping track this down. 

Odds are you have one of `rvm-prompt`, `rbenv`, `chruby` or `asdf` installed and it’s messing with your bindings. If you’re up for further sleuthing, check which of those you have installed, and we’ll see if there’s a way to work around it.

Otherwise, we you can disable the ruby segment entirely and it _should_ bypass this mess without you having to comment anything out:

    set -g theme_display_ruby no

**Comment by dwrz** [comment]
I will follow-up tomorrow (about to crash), but I have a hypothesis.

I don't know what `asdf` is, but I do have a fish alias with that name, which runs:
`setxkbmap dvorak`

I use it to switch back to Dvorak layout from Qwerty (which I switch to using `aoeu`), something I originally set up in case someone else needed to use my machine. If I run that command now, it resets my `xmodmap`.

Without looking at the code -- I'm guessing something invokes the alias, which runs `setxkbmap`, which resets my xmodmap? An easy fix on my end, which is what I want anyway, would be to have that alias run `xmodmap` after switching the layout.

Thank you for reminding me about the setting and again for all your help and patience -- feel like I could have done more work here initially, but have learned something working through this. : )

**Comment by dwrz** [comment]
Just an update:

- I don't have any of the programs you listed installed.
- Changing the alias on my end didn't fix the problem, but setting the variable for display_ruby did.
- I tried running `type -q asdf` (from line 658 of fish_prompt), and I got no error. Would `which` be more appropriate here? I know this is already an extreme edge case, but maybe someone might have `asdf` aliased to something more dangerous than keyboard mappings?
